scripts/tools/EXT_Exp_MetricProgrammable.rst

```diff
@@ -0,0 +1,120 @@
+
+    // Create metrics from metric programmable handles
+    for(uint32_t i = 0; i < programmableCount; i++){
+    
+        ${t}_metric_programmable_exp_handle_t * programmableHandle = metricProgrammableHandles[i];
+        ${t}_metric_programmable_exp_properties_t programmableProperties{};
+        ${t}MetricProgrammableGetPropertiesExp(programmableHandle, &programmableProperties);
+        
+        // Choose programmable handle of interest
+        if(strcmp(programmableProperties.name, "EU_ACTIVE" ) == 0){
+            
+            // Get Parameter descriptions
+            ${t}_metric_programmable_param_exp_desc_t * paramDescription = allocate (sizeof(${t}_metric_programmable_param_exp_desc_t) * programmableProperties.parameterCount);
+            ${t}MetricProgrammableGetParamDescriptionExp(programmableHandle, programmableProperties.parameterCount, paramDescription);
+            
+            // Get Parameter Value descriptions for the 0th parameter
+            ${t}_metric_programmable_param_value_exp_desc_t * valueDesc = allocate(sizeof(${t}_metric_programmable_param_value_exp_desc_t) * paramDescription[0].valueDescriptionCount);
+            ${t}MetricProgrammableGetParamValueDescriptionExp(programmableHandle, 0, paramDescription[0].valueDescriptionCount, valueDesc);
+            
+            // Setting value for the 0th parameter only
+            ${t}_metric_programmable_param_value_exp_t parameterValue;
+            parameterValue.value = valueDesc[0].valueDesc.ui64;
+            
+            // Create Metric
+            ${t}_metric_handle_t metricHandle{};
+            char metricName[ZET_MAX_METRIC_NAME] = "eu_active_minimum";
+            char metricDescription[ZET_MAX_METRIC_DESCRIPTION] = "eu_active_minimum_desc";
+            ${t}MetricCreateFromProgrammableExp(programmableHandle, &parameterValue, 1, metricName, metricDescription, &metricHandleCount, null_ptr);
+            metricHandles = allocate(sizeof(${t}_metric_handle_t) * metricHandleCount);
+            ${t}MetricCreateFromProgrammableExp(programmableHandle, &parameterValue, 1, metricName, metricDescription, &metricHandleCount, metricHandles);
+        }
+        
+        //Create Metric Group from metrics
+        uint32_t handlesSize = metricHandlesUsed.size();
+        char metricGroupName[ZET_MAX_METRIC_GROUP_NAME] = "eu_active";
+        char metricGroupDescription[ZET_MAX_METRIC_GROUP_DESCRIPTION] = "eu_active_desc";
+        ${t}MetricGroupCreateExp(device, metricGroupName, metricGroupDescription, &metricGroup);
+        ${t}MetricGroupAddMetricExp(metricGroup, &metricHandlesUsed[0], null_ptr, null_ptr);
+    }
+    
+    //Activate Metric group
+    //Collect Metric group using available sampling types
+    
+    //Cleanup
+    ${t}MetricGroupDestroyExp(metricGroup);
+    
+    for(uint32_t j = 0; j < metricHandleCount; j++){
+        ${t}MetricDestroyExp(metricHandles[j]);
```

Comment:
MetricDestroyFromProgrammable (or some other name) for symmetry?

Reply:
Like mentioned above, 
"The idea is that metrics could be created from different mechansims but destroy to be applicable for all created metrics.
So did not want to have the creation method added to the destroy name."